opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectSimulationResult.jsx

```diff
@@ -118,24 +182,52 @@ const StixCoreObjectSimulationResult = ({ id, type }) => {
   const classes = useStyles();
   const [open, setOpen] = useState(false);
   const [openCallToAction, setOpenCallToAction] = useState(false);
+  const { oBasConfigured, oBasDisableDisplay } = useXTM();
   const isEnterpriseEdition = useEnterpriseEdition();
   const { enabled, configured } = useAI();
-  const { oBasConfigured, oBasDisableDisplay } = useXTM();
+  const isSimulatedEmailsAvailable = enabled && configured && isEnterpriseEdition;
   const [simulationType, setSimulationType] = useState('technical');
+  const [platforms, setPlatforms] = useState([{ label: 'Windows', value: 'Windows' }]);
+  const [architecture, setArchitecture] = useState('x86_64');
   const [selection, setSelection] = useState('random');
   const [interval, setInterval] = useState(2);
-  const [useGenAI, setUseGenAI] = useState(false);
   const [isSubmitting, setIsSubmitting] = useState(false);
   const [result, setResult] = useState(null);
   const [resultError, setResultError] = useState(null);
   const [filters, helpers] = useFiltersState(emptyFilterGroup);
   const { t_i18n } = useFormatter();
   const isGrantedToUpdate = useGranted([KNOWLEDGE_KNUPDATE]);
 
+  // Determine the query based on the type
+  let attackPatternsQuery;
+  if (type === 'container') {
+    attackPatternsQuery = StixCoreObjectSimulationResultAttackPatternsForContainersQuery;
+  } else if (type === 'threat') {
+    attackPatternsQuery = stixCoreObjectSimulationResultAttackPatternsForThreatsQuery;
+  } else {
+    throw new Error('Type of the simulation should be container or threat');
+  }
+
+  // Fetch the attackPatterns using the selected query
+  const attackPatterns = useLazyLoadQuery(attackPatternsQuery, { id });
```

Comment:
We recomand to use useQueryLoading instead of useLazyLoadQuery... Do you think we can keep lazy load heare @aHenryJard ?

Reply:
I'm gonna replace it myself @savacano28